### PR TITLE
Support of golang types Time, Uint and Uint64 added

### PR DESCRIPTION
**Pull request by hus787**
Minor changes: iota used instead of defining const values by hand

### CODE REVIEW

File: type.go [hunk 1]
```diff
@@ -12,35 +12,40 @@ type OType byte
...
-	LINKMAP      OType = 16
-	BYTE         OType = 17
-	TRANSIENT    OType = 18
-	DATE         OType = 19
-	CUSTOM       OType = 20
-	DECIMAL      OType = 21
-	LINKBAG      OType = 22
-	ANY          OType = 23
-	UNKNOWN      OType = 255 // driver addition
+	BOOLEAN OType = iota
```
Comment (dennwc) [inline type.go]: As a general rule, you should not use iota to reflect constants defined in third-party library.

File: type.go [hunk 2]
```diff
@@ -12,35 +12,40 @@ type OType byte
...
+)
+const (
+	UNKNOWN OType = 255 // driver addition
 )
 
 // detect the int size (4 or 8 bytes)
-const intSize = unsafe.Sizeof(int(0))
+const (
+	intSize  = unsafe.Sizeof(int(0))
+	uintSize = unsafe.Sizeof(uint(1))
```
Comment (dennwc) [inline type.go:45]: Why to use both? Is it possible to hit the case when `intSize != uintSize`?

Comment (hus787) [inline type.go:45]: Since the size is implementation detail, then it's better to play safe.

File: type.go (hunk 1 above)
Comment (hus787) [inline type.go]: Agreeed. Shall revert.

Wish we could do something like: https://github.com/lib/pq/blob/1046c2c577f5be9d829bda808a5696c4c74c9f3b/oid/gen.go#L51

File: type.go (hunk 2 above)
Comment (dennwc) [inline type.go:45]: I'd better to add test for this extra-rare case if you like :) It's architecture-dependent, and I'm, not aware of any architectures with different sizes for uint and int.

**Comment by dennwc** [comment]
I have fixed CI, it should work now.
When all comments will be addressed, please squash your commits and rebase it on top of current v2 branch.

**Comment by dennwc** [comment]
Merged as a1fc07b5a7e42c45d849702d0f3c50c2e0f60a1d

**Pull request closed without merging**